at_secondary/at_secondary_server/lib/src/verb/handler/notify_fetch_verb_handler.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:collection';
+import 'dart:convert';
+
+import 'package:at_commons/at_commons.dart';
+import 'package:at_persistence_secondary_server/at_persistence_secondary_server.dart';
+import 'package:at_secondary/src/verb/handler/abstract_verb_handler.dart';
+import 'package:at_secondary/src/verb/verb_enum.dart';
+import 'package:at_server_spec/at_server_spec.dart';
+import 'package:at_server_spec/at_verb_spec.dart';
+
+class NotifyFetchVerbHandler extends AbstractVerbHandler {
+  static NotifyFetch notifyFetch = NotifyFetch();
+
+  NotifyFetchVerbHandler(SecondaryKeyStore keyStore) : super(keyStore);
+
+  @override
+  bool accept(String command) =>
+      command.startsWith('${getName(VerbEnum.notify)}:fetch');
+
+  @override
+  Verb getVerb() {
+    return notifyFetch;
+  }
+
+  @override
+  Future<void> processVerb(
+      Response response,
+      HashMap<String, String?> verbParams,
+      InboundConnection atConnection) async {
+    var notificationId = verbParams['notificationId'];
+    var atNotification =
+        await AtNotificationKeystore.getInstance().get(notificationId);
+    if (atNotification == null) {
+      response.data = jsonEncode(
+          {'id': notificationId, 'status': NotificationStatus.expired.name});
+      return;
+    }
+    response.data = _getJsonEncodedString(atNotification);
+  }
+
+  String _getJsonEncodedString(AtNotification atNotification) {
+    var atNotificationJson = atNotification.toJson();
+    for (MapEntry entry in atNotificationJson.entries) {
+      if (entry.value is! String) {
+        atNotificationJson[entry.key] = entry.value.toString();
+      }
+    }
+    return jsonEncode(atNotificationJson);
```

Comment:
Why all of the above? Looking at https://api.flutter.dev/flutter/dart-convert/jsonEncode.html, can't we just immediately `return jsonEncode(atNotificationJson);` ?

Reply:
@gkc :  AtNotification contains fields that are not converted to JSON by default; For example fields of type DateTime and enum's. So, we need to extend logic to have toJson on all  enums and DateTime fields and then use `jsonEncode(atNotificationJson)`.

Will make the necessary changes required and keep you updated.